pdblp/pdblp.py

```diff
@@ -731,6 +745,44 @@ def bsrch(self, domain):
                     data.append(f.getElementAsString("StringValue"))
         return pd.DataFrame(data)
 
+    def secf(self, query, max_results=10, yk_filter="NONE"):
+        """
+        This function uses the Bloomberg API to retrieve Bloomberg
+        SECF Data queries. Returns list of tickers.
+
+        Parameters
+        ----------
+        query: string
+            A character string representing the desired query. Example "IBM"
+        max_results: int
+            Maximum number of results to return. Default 10. 
+        yk_filter: string
+            A character string respresenting a Bloomberg yellow-key to limit
+            search results to. Valid values are: CMDT, EQTY, MUNI, 
+            PRFD, CLNT, MMKT, GOVT, CORP, INDX, CURR, MTGE. Default NONE.
+
+        Returns
+        -------
+        data: list
+            List of bloomberg tickers from the SECF function
+        """
+        logger = _get_logger(self.debug)
+        request = self.instrService.createRequest("instrumentListRequest")
+        request.set("query", query)
+        request.set("maxResults", max_results)
+        request.set("yellowKeyFilter", "YK_FILTER_%s" % yk_filter)
```

Comment:
I think to be consistent with how parameters are set elsewhere in the library, this convenience string concatenation should not be done for the user, e.g. the user should pass `YK_FILTER_EQTY` and this line should be changed to `request.set("yellowKeyFilter", yk_filter)`. Also while it does appear that `YK_FILTER_NONE` is supported I think it would be better stylistically to have yk_filter=None as a default parameter and then
if yk_filter:
    request.set("yellowKeyFilter", yk_filter)